content/blog/2019/06/2019-06-06-cloudfeature-gsoc-cp1.adoc

```diff
@@ -0,0 +1,104 @@
+
+You have a user data file called ```jenkins-bootstrap.tpl```, like this
+```
+...
+# all dependencied installed
+sudo yum update -y
+sudo yum install jenkins
+sudo systemctl start jenkins.service
+sudo yum install -y amazon-efs-utils
+# then you mount the EFS
+
+sudo mount -t ${EFS_ID}:/ ${MOUNT_POINT}
+```
+Then you create a ec2 instance with such statement
+```
+resource "aws_instance" "jenkins" {
+  ami           = "${data.any_ami_you_want}"
+  instance_type = "t2.micro"
+  user_date     = 
+  tags = {
+    Name = "HelloWorld"
+  }
+}
+
+
+data "template_file" "jenkins_init" {
+  template = "${file("${path.module}/jenkins-bootstrap.tpl")}"
+  vars = {
+    EFS_ID        = "${aws_efs_file_system.efs_jenkins.id}"
+    MOUNT_POINT   = "${var.mount_point}"
+    
+  }
+}
+```
+
+As you can see, user needn't know the efs-id of the created system, just build a dependency like this, and everything is set up automatically. 
+
+3. No convenience is given to user, even we implement the cloud management. We cannot give more feature than CasC(configuration as code) tools and we won't need to. And if we only implement EFS related CasC, what about other resources who also might cooperate with EFS ? e.g. There might not only be one consumer using the EFS, if some service wants to pull information from the Jenkins workspace, that would be inconvenient.
+
+So you can see, the user can do the exactly same thing we want to implement, using other tools and current EWM, possibly in a better way. Then we find that, though I personally did quite a lot to this idea, like learning AWS and Jenkins plugin developement and some working codes, we have to admit our idea might not do good to the community.
+
+=== Done and TODOs
+This is what we've already done:
+
+. The Web UI to define a AWS EFS disk
+. Implemented a demo workflow code
+. Implemented the AWS EFS utils, like create/destroy/mount an EFS.
+
+And unfortunately we have to say bye-bye to those dones. We realize sometimes negative results are also results. A good programmer should know when he is wrong, and has the courage to admit it. Honestly, we were a little bit uneasy when we did the decision, but it doesn't matter, we will continue and spend time on something else.
+
+=== Plan for the future
+
+I previously worked with the JCasC compatiblity of EWM, and my lead mentor Martin, is also interested in JCasC. Then we talked with Oleg and agree JCasC would be a good project to work on. Now we want to have a meeting to decide the priorities of which plugins should we work on first. Also, JCasC is somehow the extension of cloud features, as CasC is important in managing large clusters, if we do more work on JCasC, the integrations of Jenkins and clouds would also be promoted. And that is what we want to see.
+   
+
```

Comment:
It would be great to add some details about meetings once we have a stakeholder meeting